### PR TITLE
Use cluster scope for NodeNetworkState

### PR DESCRIPTION
**Pull request by qinqon**
Followed instructions at [1]

https://github.com/operator-framework/operator-sdk/blob/master/doc/operator-scope.md

Signed-off-by: Quique Llorente <quique.llorente@gmail.com>

### CODE REVIEW

File: deploy/role.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 apiVersion: rbac.authorization.k8s.io/v1
-kind: Role
+kind: ClusterRole
```
Comment (phoracek) [inline deploy/role.yaml]: We have to be more strict here.

Please use cluster-role only where we needed (AFAIK monitoring Nodes and handling NodeNetworkState).

The rest should be limited to our namespace (we should not have rights to remove all existing objects).

Finally, I'm sure we don't need all the resource that are listed here, but that is for another PR.

Comment (qinqon) [reply deploy/role.yaml]: Done, let's do cleanup after we merge to master with all the changes so we know what we access.

Comment (qinqon) [reply deploy/role.yaml]: It's taking me time to put the correct rules at the two places.

Comment (qinqon) [reply deploy/role.yaml]: I needed to put "pods" and "pods/exec" at ClusterRole, maybe after the distributed handler we can change that, at least now is more restrictive.

Comment (phoracek) [reply deploy/role.yaml]: weird. but it won't be needed in the final version with shared containre between nmstatectl and operator, so it's fine.

**Comment by phoracek** [comment]
ci test please

**Pull request merged**